HDF5Examples/CMakeLists.txt

```diff
@@ -99,53 +99,40 @@ if (H5_HAVE_PARALLEL)
   INCLUDE_DIRECTORIES (${MPI_C_INCLUDE_DIRS})
 endif ()
 
-# Determine if a threading package is available on this system
-option (HDF5_ENABLE_THREADS "Enable thread support" ON)
-set (THREADS_PREFER_PTHREAD_FLAG ON)
-find_package (Threads)
-if (Threads_FOUND)
-  set (H5_HAVE_THREADS 1)
-  set (CMAKE_REQUIRED_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
-
-  # Determine which threading package to use
-  # Comment out check for C11 threads for now, since it conflicts with the
-  # current --std=c99 compile flags at configuration time.  When we switch to
-  # --std=c11, this can be uncommented.
-  #CHECK_INCLUDE_FILE("threads.h" HAVE_THREADS_H)
-  if (WIN32)
-    # When Win32 is available, we use those threads
-    set (H5_HAVE_WIN_THREADS 1)
-  elseif (HAVE_THREADS_H)
-    # When C11 threads are available, those are the top choice
-    set (H5_HAVE_C11_THREADS 1)
-  elseif (CMAKE_USE_PTHREADS_INIT)
-    set (H5_HAVE_PTHREAD_H 1)
-  else ()
-    message (FATAL_ERROR " **** thread support requires C11 threads, Win32 threads or Pthreads **** ")
+#-----------------------------------------------------------------------------
```

Comment:
This piece seems wrong, as it's eliminating all the macros that were previously set that show what kind of threads are found.

Reply:
No that section is not needed for Examples. These changes only affect the Examples when using the library binary.

Reply:
OK, I see.

Then this entire section of the CMake file should be removed, since the examples don't use threading.

Reply:
Until we add examples that test such feature.

Reply:
If the library requires the THREADS library it still needs to be checked.

Reply:
Why?   Applications that link to a threadsafe version of HDF5 don't need to make this check in *their* CMake builds, correct?